### PR TITLE
feat: Update task definition without deployment

### PR DESCRIPTION
**Pull request by pssaravanan**
Changes are done as part of this p.r:
* create_task_defn
* update_task_defn
* ECRClient  class extraction to build and push image

### CODE REVIEW

**Review by shoan — CHANGES_REQUESTED** [review]
The upload to ecr functionality is broken in this branch. Cloudlift can't find the local image.

File: cloudlift/__init__.py [hunk 1]
```diff
@@ -113,6 +117,30 @@ def deploy_service(name, environment, version, build_arg):
...
 
+@cli.command()
+@_require_environment
+@_require_name
+@click.option('--version', default=None,
+              help='local image version tag')
+@click.option("--build-arg", type=(str, str), multiple=True, help="These args are passed to docker build command "
+                                                                  "as --build-args. Supports multiple.\
+                                                                   Please leave space between name and value" )
+def create_task_defn(name, environment, version, build_arg):
```
[the remark this answers is not in the document]
Comment (pssaravanan) [reply cloudlift/__init__.py]: Made the changes

File: cloudlift/deployment/service_updater.py [hunk 2]
```diff
@@ -40,26 +43,28 @@ def run(self):
...
         ecs_client = EcsClient(None, None, self.region)
 
         jobs = []
         for index, service_name in enumerate(self.ecs_service_names):
             log_bold("Starting to deploy " + service_name)
             color = DEPLOYMENT_COLORS[index % 3]
-            image_url = self.ecr_image_uri
-            image_url += (':' + self.version)
+            image_url = ecr_client.ecr_image_uri
+            print(image_url)
```
[the remark this answers is not in the document]
Comment (pssaravanan) [reply cloudlift/deployment/service_updater.py]: Removed the print. It was used for debugging. missed to remove.

File: cloudlift/deployment/task_definition_service.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+from cloudlift.config.logging import log_bold, log_intent, log_warning
+from cloudlift.deployment import EcrClient, UnrecoverableException, EcsClient, DeployAction, EcsTaskDefinition
+from cloudlift.deployment.deployer import build_config, print_task_diff
+
+
+def _complete_image_url(ecr_client: EcrClient):
+    return ecr_client.ecr_image_uri + ':' + ecr_client.version
+
+
+class TaskDefinitionService:
```
[the remark this answers is not in the document]
Comment (pssaravanan) [reply cloudlift/deployment/task_definition_service.py]: Made the changes

**Review by shoan — APPROVED** [review]

**Pull request merged**